kotlin-extensions/src/test/java/com/amplifyframework/ktx/DataStoreCategoryExtTest.kt

```diff
@@ -0,0 +1,99 @@
+
+import com.amplifyframework.core.Amplify
+import com.amplifyframework.core.AmplifyConfiguration
+import com.amplifyframework.core.Consumer
+import com.amplifyframework.datastore.DataStoreItemChange
+import com.amplifyframework.datastore.DataStoreItemChange.Initiator.LOCAL
+import com.amplifyframework.datastore.DataStoreItemChange.Type.CREATE
+import com.amplifyframework.datastore.DataStorePlugin
+import com.amplifyframework.testmodels.commentsblog.BlogOwner
+
+import kotlinx.coroutines.runBlocking
+import org.json.JSONObject
+
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.`when`
+import org.mockito.Mockito.any
+import org.mockito.Mockito.eq
+import org.mockito.Mockito.mock
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class DataStoreCategoryExtTest {
+    private lateinit var mockPlugin: DataStorePlugin<*>
+    private lateinit var blogOwner: BlogOwner
+    private lateinit var values: MutableList<BlogOwner>
+
+    @Before
+    fun setupMockDataStorePlugin() {
+        // When items are saved, store them in a local buffer.
+        values = mutableListOf()
+
+        // We'll mock away the plugin.
+        mockPlugin = mock(DataStorePlugin::class.java)
+
+        // Prepare an item to save.
+        blogOwner = BlogOwner.builder()
+            .name("Tony Danielsen, J.D.")
+            .build()
+
+        `when`(mockPlugin.pluginKey).thenReturn("fakeDataStorePlugin")
+
+        // When its saved, fire success on the plugin with a DataStoreItemChange.
+        // Also save the the item to the buffer.
+        `when`(mockPlugin.save(eq(blogOwner), any(), any())).thenAnswer {
+            val onResult = it.getArgument<Consumer<DataStoreItemChange<BlogOwner>>>(1)
+            val value = it.getArgument<BlogOwner>(0)
+            values.add(value)
+            val change = DataStoreItemChange.builder<BlogOwner>()
+                .initiator(LOCAL)
+                .item(value)
+                .itemClass(BlogOwner::class.java)
+                .type(CREATE)
+                .build()
+            onResult.accept(change)
+        }
+
+        // When the plugin is queried, fake a response by just returning
+        // the saved list.
+        `when`(mockPlugin.query(eq(BlogOwner::class.java), any(), any())).thenAnswer {
+            val onResult = it.getArgument<Consumer<Iterator<BlogOwner>>>(1)
+            onResult.accept(values.iterator())
+        }
+
+        // Okay, now add the plugin.
+        Amplify.addPlugin(mockPlugin)
+        val config = AmplifyConfiguration.fromJson(JSONObject())
+        Amplify.configure(config, getApplicationContext())
+    }
+
+    @Test
+    fun highlightDeveloperExperience() {
+        runBlocking {
+            Amplify.DataStore.save(blogOwner)
+            val results = Amplify.DataStore.query(BlogOwner::class.java)
```

Comment:
We're going to have to start generating Kotlin models, too. We'll need to add a "which language?" prompt to the CLI.

Reply:
What makes you say we will "have" to? Not that I necessarily disagree, just curious

Reply:
"Have" was the wrong word. "Should," I mean. I think they'd make more sense as data classes.